Xdebug on CLI and inside WSL

rabauss opened the pull request:
Here are some updates for xdebug especially on CLI and inside WSL

* Update debugging cli commands  
* Unify toggling xdebug tooling
* Add configure host IP in wsl

rabauss commented on docs-3.x/guides/setup-lando-on-windows-with-wsl-2.md:
````diff
@@ -70,6 +70,44 @@ You'll want to do all of your shell activity from within the WSL system, and you
+```bash
+# Set correct dev host IP to Windows
+export LANDO_HOST_IP_DEV=host.wsl.internal
+sudo sed -i "/$LANDO_HOST_IP_DEV/d" /etc/hosts && sudo sh -c "echo $(cat /etc/resolv.conf | grep nameserver | awk '{print $2; exit;}') $LANDO_HOST_IP_DEV >> /etc/hosts"
````
Today we realized that this seems to be a bad solution. Maybe only the following is the solution and you always need to to a `lando rebuild` if the IP of wsl or of the wsl host has changed 😢 
```suggestion
export LANDO_HOST_IP_DEV=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2; exit;}')
```

rabauss commented on docs-3.x/guides/setup-lando-on-windows-with-wsl-2.md:
```diff
@@ -70,6 +70,55 @@ You'll want to do all of your shell activity from within the WSL system, and you
+// ~/.bashrc
+# Set correct dev host to Windows
+export LANDO_HOST_NAME_DEV=host.wsl.internal
+export LANDO_HOST_GATEWAY_DEV=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2; exit;}')
```
Unfortunately I don't have a better solution - in particular you need to rebuild if wsl changes windows host ip address! Are there any better suggestions to get the windows host ip?

rabauss commented:
@pirog  Are there any suggestions to merge this PR?

reynoldsalec commented:
@rabauss sorry this has sat around for a while, it looks like there might be some very cool changes!

Please note that locations of guides has drifted a bit, they're still in this repo (most of the docs have moved to separate plugin repos), but if you wouldn't mind updating your PR for the new location, that would be a big help.

rabauss commented:
@reynoldsalec I just rebased my feature branch to the current `main` and fixed some conflicts.
Now the suggested changes should be ready to merge again.

reynoldsalec commented on guides/lando-phpstorm.md:
```diff
@@ -60,26 +61,47 @@ export XDEBUG_SESSION="PHPSTORM"
-      - docker-php-ext-enable xdebug
-      - pkill -o -USR2 php-fpm
-      - echo "Xdebug enabled"
+    cmd: rm -f /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini && docker-php-ext-enable xdebug && pkill -o -USR2 php-fpm && echo "Xdebug enabled"
```
@rabauss is there a reason it's necessary to do everything in the single command? Seems easier to read in the YML array format.

rabauss replied:
I just wanted to make the duplicate toolings unique. But I think one line is faster because it's executed in one step, isn't it?

reynoldsalec replied:
Did some testing with `time` and you're right, running them as a single command is faster. It's not a big difference (maybe ~.25s or less on average), but somewhat perceptible. Let's go with your suggestion!

reynoldsalec reviewed (commented):
Just one minor question/comment, otherwise I think this is ready to go, thanks @rabauss!
